src/python/joint_cell_plotting.py

```diff
@@ -0,0 +1,161 @@
+#!/usr/bin/env python3
+
+"""
+This script QCs barcodes via ATAC frags & TSS and RNA UMIs & genes,
+and plots all barcodes colored by joint QC status. It also generates the 
+same plot with transparency added to show density. 
+"""
+
+import argparse
+import logging
+import numpy as np
+import pandas as pd
+from plotnine import *
+
+def parse_arguments():
+    parser = argparse.ArgumentParser(description="Plot barcodes by RNA and ATAC QC status")
+    parser.add_argument("pkr", help="PKR name")
+    parser.add_argument("rna_metrics_file", help="Filename for RNA metrics tsv matrix")
+    parser.add_argument("atac_metrics_file", help="Filename for ATAC metrics tsv file")
+    parser.add_argument("min_UMIs", type=int, help="Cutoff for minimum number of UMIs")
+    parser.add_argument("min_genes", type=int, help="Cutoff for minimum number of genes")
+    parser.add_argument("min_TSS", type=int, help="Cutoff for minimum TSS score")
+    parser.add_argument("min_frags", type=int, help="Cutoff for minimum number of ATAC fragments")
+    parser.add_argument("barcode_metadata_file", help="Filename for barcode metadata csv file")
+    
+    return parser.parse_args()
+
+def get_split_lines(file_name, delimiter):
+    with open(file_name, "r") as f:
+        for line in f:
+            yield line.rstrip().split(sep=delimiter)
+
+def merge_dicts(dict_1, dict_2):
+    """Merge dictionaries by key; combine values into quadruple, fill with 0s if key not in both dicts"""
+    keys = set(dict_1.keys() | dict_2.keys())
+    merged = {k: (dict_1.get(k, (0,0)) + dict_2.get(k, (0,0))) for k in keys}
+    
+    return(merged)
+
+def get_metrics(rna_metrics_file, atac_metrics_file):
+    """Read files and aggregate metrics into Pandas dataframe"""
+    rna_metrics_contents = get_split_lines(rna_metrics_file, delimiter="\t")
+    UMIs = []
+    genes = []
+    rna_barcodes = []
+    next(rna_metrics_contents) # skip header
+    for line in rna_metrics_contents:
+        if int(line[1]) >= 10: # impose cutoff of 10 UMIs; RNA metrics file contains all values
```

Comment:
This is not 10 but min_UMI right?

Reply:
if not, it stills need to be parametrized

Reply:
The Seurat barcode metadata output tsv includes all barcodes, not just those that have at least 10 UMIs (i.e. passing minimum cutoff value). The ArchR barcode metadata output has a cutoff of 10 frags, so I wanted to add the 10 UMI cutoff to the RNA so that the plot would look nice. Should I go into the Seurat script and change the output myself?

Reply:
why couldn't you just use the min_UMI flag?

Reply:
I'm just using this cutoff to get rid of the barcodes that have 1-9 UMIs, as plotting those looks weird when there are no corresponding points in the 1-9 range of the frags axis.
![joint_cell_calling_ugly](https://user-images.githubusercontent.com/87831251/199582230-a713c11f-514b-4c3b-8d2c-519033d97d53.png)

Reply:
than just parametrize it. set it to 10 by default and write a comment about why that is needed

Reply:
This might be caused because of how the pipeline works and it is run. Don't change it in Seurat.